sphinx/tutorial/cylc/furthertopics/clock-triggered-tasks.rst

```diff
@@ -0,0 +1,179 @@
+
+.. code-block:: cylc
+
+   [cylc]
+       UTC mode = True # Ignore DST
+
+   [scheduling]
+       initial cycle point = TODO
+       final cycle point = +P1D # Run for one day
+       [[dependencies]]
+           [[[PT1H]]]
+               graph = bell
+
+   [runtime]
+       [[root]]
+           [[[events]]]
+               mail events = failed
+       [[bell]]
+           env-script = eval $(rose task-env)
+           script = printf 'bong%.0s\n' $(seq 1 $(cylc cyclepoint --print-hour))
+
+Change the initial cycle point to 00:00 this morning (e.g. if it was
+the first of January 2000 we would write ``2000-01-01T00Z``).
+
+We now have a simple suite with a single task that prints "bong" a number
+of times equal to the (cycle point) hour.
+
+Run your suite using::
+
+   cylc run clock-trigger
+
+Stop the suite after a few cycles using the :guilabel:`stop` button in the
+``cylc gui``. Notice how the tasks run as soon as possible rather than
+waiting for the actual time to be equal to the cycle point.
+
+
+Clock Triggering Tasks
+----------------------
+
+We want our clock to only ring in real time rather than the simulated
+cycle time.
+
+To do this, add the following lines to the ``[scheduling]`` section of
+your ``suite.rc``:
+
+.. code-block:: cylc
+
+   [[special tasks]]
+       clock-trigger = bell(PT0M)
+
+This tells the suite to clock trigger the ``bell`` task with a cycle
+offset of ``0`` hours.
+
+Save your changes and run your suite::
+
+   cylc run clock-trigger
+
+
+Results
+-------
+
+Your suite should now be running the bell task in realtime. Any cycle times
+that have already passed (such as the one defined by ``initial cycle time``)
+will be run as soon as possible, while those in the future will wait for that
+time to pass.
+
+At this point you may want to leave your suite running until the next hour
+has passed in order to confirm the clock triggering is working correctly.
+Once you are satisfied, stop your suite.
+
+By making the ``bell`` task a clock triggered task we have made it run in
+realtime. Thus, when the time caught up with the cycle time, the bell task
+triggered.
+
+
+Further Clock Triggering
+------------------------
+
+We will now modify our suite to run tasks at quarter-past, half-past and
+quarter-to the hour.
+
+Open your ``suite.rc`` and modify the ``[runtime]`` section by adding the
+following:
+
+.. code-block:: cylc
+
+   [[quarter_past, half_past, quarter_to]]
+       script = echo 'chimes'
+
+Similarly, modify the ``[[scheduling]]`` section as follows:
+
+.. code-block:: cylc
+
+   [[special tasks]]
+       clock-trigger = bell(PT0M), quarter_past(PT15M), half_past(PT30M), quarter_to(PT45M)
+   [[dependencies]]
+       [[[PT1H]]]
+           graph = bell => quarter_past => half_past => quarter_to
+
+Note the different values used for the cycle offsets of the clock-trigger tasks.
+
+Save your changes and run your suite using::
+
+   cylc run clock-trigger now
```

Comment:
Readers have already been instructed as to how to run the suite above (twice at the moment but once if lines 107-109 are amended) so consider instead stating something like 'save...and run your suite, appending the argument ``now`` to the standard command'. Better to try and get the reader to recall and apply what they have just learnt, and avoids repetition.

Reply:
With the following note I think this is OK. I have removed the previous occurrence.